src/character/import.js

```diff
@@ -171,8 +171,8 @@ export async function getCharacterData(characterId, syncId) {
     body['updateId'] = syncId;
   }
 
-  return new Promise((resolve, reject) => {
-    fetch(`${parsingApi}/proxy/character`, {
+  try {
```

Comment:
What's your reasoning for changing this?

Reply:
Yeah, to be honest I think most people don't actually fully understand promises properly in js 😅. I'm planning on doing a talk on them at some point.

Promises were added to JS as third party ways of avoiding "callback hell" but were added into core with a few performance boosts but they didn't require any new syntax, they also had their own issues. `async`/`await` was added not too long after which added new syntax that abstracted some stuff out of the way (I believe using generator functions which had been added prior too). Generally nowadays I generally recommend to people that they use `async`/`await` whereever possible and not mix and match with `.then` as the former is generally a little easier to read (I guess once you wrap your head around how it works). I tend to only use `new Promise` when I need to pass the resolver/rejector to something else and generally try to avoid `.then` where possible as `.then` loses the stacktrace too where as `async`/`await` does not.

Some videos that might be useful or just if you're curious:
https://www.youtube.com/watch?v=4GpwM8FmVgQ
https://www.youtube.com/watch?v=Em2jqwROdZc

Reply:
Thankyou!